components/logins/ios/Logins/LoginsStorage.swift

```diff
@@ -56,6 +64,29 @@ open class LoginsStorage {
         return self.raw
     }
 
+    private func doOpen(_ key: String) throws {
+        if self.raw != 0 {
+            return
+        }
+
+        self.raw = try LoginsStoreError.unwrap({ err in
+            sync15_passwords_state_new(self.dbPath, key, err)
+        })
+
+        do {
+            self.interrupt_handle_lock.lock()
```

Comment:
Doesn't this need its own instance of `NSLock` that isn't also used by `interrupt()`? Otherwise, wouldn't the `interrupt()` function be blocked waiting for this to complete? Maybe @garvankeeley can shed some light on this.

Reply:
This lock is only to ensure that writes and reads to `interrupt_handle` have synchronization. It needs to be held in anything that reads or writes the interrupt handle, otherwise we'd have a data race (since swift properties are not atomic).

Reply:
Ohh, I see. Nevermind. The long-running operation executes before this. I was worried that the lock would prevent the interrupt from ever running.

Reply:
Yeah, it's annoying we still need locking here, the property could be atomic if swift had such a thing...